pims/cine.py

```diff
@@ -68,147 +79,204 @@ def _build_struct(dtype):
-    ('bi_bit_count', WORD),
-    ('bi_compression', DWORD),
-    ('bi_image_size', DWORD),
-    ('bi_x_pels_per_meter', LONG),
-    ('bi_y_pels_per_meter', LONG),
-    ('bi_clr_used', DWORD),
-    ('bi_clr_important', DWORD),
+    ('bi_size', UINT32),
+    ('bi_width', INT32),
+    ('bi_height', INT32),
+    ('bi_planes', UINT16),
+    ('bi_bit_count', UINT16),
+    ('bi_compression', UINT32),
+    ('bi_image_size', UINT32),
+    ('bi_x_pels_per_meter', INT32),
+    ('bi_y_pels_per_meter', INT32),
+    ('bi_clr_used', UINT32),
+    ('bi_clr_important', UINT32),
 ]
 
+# Inserted Res_ prefix whenever field is obsolete (to easy filter)
 SETUP_FIELDS = [
-    ('frame_rate_16', WORD),
-    ('shutter_16', WORD),
-    ('post_trigger_16', WORD),
-    ('frame_delay_16', WORD),
-    ('aspect_ratio', WORD),
-    ('contrast_16', WORD),
-    ('bright_16', WORD),
-    ('rotate_16', BYTE),
-    ('time_annotation', BYTE),
-    ('trig_cine', BYTE),
-    ('trig_frame', BYTE),
-    ('shutter_on', BYTE),
-    # Guessed at length... because it isn't documented!  This seems to work.
-    ('description_old', '121s'),
+    ('Res_frame_rate_16', UINT16),
+    ('Res_shutter_16', UINT16),
+    ('Res_post_trigger_16', UINT16),
+    ('Res_frame_delay_16', UINT16),
+    ('aspect_ratio', UINT16),
+    ('Res_7', UINT16),
+    ('Res_8', UINT16),
+    ('Res_9', UINT8),
+    ('Res_10', UINT8),
+    ('Res_11', UINT8),
+    ('trig_frame', UINT8),
+    ('Res_12', UINT8),
+    ('description_short', '121s'),
     ('mark', '2s'),
-    ('length', WORD),
-    ('binning', WORD),
-    ('sig_option', WORD),
-    ('bin_channels', SHORT),
-    ('samples_per_image', BYTE)] + \
-    [('bin_name%d' % i, '11s') for i in range(8)] + [
-        ('ana_option', WORD),
-        ('ana_channels', SHORT),
-        ('res_6', BYTE),
-        ('ana_board', BYTE)] + \
-    [('ch_option%d' % i, SHORT) for i in range(8)] + \
-    [('ana_gain%d' % i, FLOAT) for i in range(8)] + \
-    [('ana_unit%d' % i, '6s') for i in range(8)] + \
-    [('ana_name%d' % i, '11s') for i in range(8)] + [
-    ('i_first_image', LONG),
-    ('dw_image_count', DWORD),
-    ('n_q_factor', SHORT),
-    ('w_cine_file_type', WORD)] + \
-    [('sz_cine_path%d' % i, '65s') for i in range(4)] + [
-    ('b_mains_freq', WORD),
-    ('b_time_code', BYTE),
-    ('b_priority', BYTE),
-    ('w_leap_sec_dy', DOUBLE),
-    ('d_delay_tc', DOUBLE),
-    ('d_delay_pps', DOUBLE),
-    ('gen_bits', WORD),
-    ('res_1', INT16),  # Manual says INT, but this is clearly wrong!
-    ('res_2', INT16),
-    ('res_3', INT16),
-    ('im_width', WORD),
-    ('im_height', WORD),
-    ('edr_shutter_16', WORD),
-    ('serial', UINT),
-    ('saturation', INT),
-    ('res_5', BYTE),
-    ('auto_exposure', UINT),
+    ('length', UINT16),
+    ('Res_13', UINT16),
+    ('sig_option', UINT16),
+    ('bin_channels', INT16),
+    ('samples_per_image', UINT8),
+    ('bin_name', ''.join(8*['11s'])),
+    #[('bin_name{:d}'.format(i), '11s') for i in range(8)] + [
```

Comment:
You understand it correctly, this is a only a suggestion, that consciously violates backwards compatibility. It would be easy, however and with that initial unpacking, to add an option/private method to 'unzip' those vectors and recover the old behaviour (bin_name is concerned, as well as others).

Reply:
Ok, I just rendered the code uniform, back to the previous behavior for those fields.